### ISSUE TITLE
Minor Glitch: CKEditor "placeholder" clashes with LinkCommand

### ISSUE THREAD

**Issue by mmichaelis** [ckeditor/ckeditor5#9776]
## 📝 Provide detailed reproduction steps (if any)

1. Enable `placeholder` in your CKEditor configuration. See [Editor placeholder - CKEditor 5 Documentation](https://ckeditor.com/docs/ckeditor5/latest/features/editor-placeholder.html).
2. Clear the text, so that you see the placeholder.
3. Press "Link" button in toolbar.

### ✔️ Expected result

Just as for other inline styles (e.g. Bold), the text area is cleared and the link-popup appears.

### ❌ Actual result

The placeholder text is not removed, the cursor placed at the end of the placeholder text and the link-popup appears:

![ckeditor5-linkcommand-vs-empty-text](https://user-images.githubusercontent.com/71911/119942896-a0de8480-bf92-11eb-9fc5-c00622371122.png)

Fortunately, if you set a link and submit `LinkCommand` form view, the placeholder text is gone and the link is written to the text area as expected.

## 📃 Other details

* Browser: Chrome 90.0.4430.212
* OS: Windows 10
* First affected CKEditor version: 24.0.0
* Installed CKEditor plugins: Link

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by FilipTokarski** [comment]
Hi, thanks for the report. I tried this on the [docs page](https://ckeditor.com/docs/ckeditor5/latest/features/editor-placeholder.html) and also locally but I was unable to reproduce this fully. Placeholder text indeed does not disappear, but the selection is always at the beginning of a paragraph:

https://user-images.githubusercontent.com/34380544/120005739-6e577a80-bfd8-11eb-956b-bcafd857c3c0.mp4

Maybe apart from link plugin, you use some custom plugins or some specific plugins combination causes this?

**Comment by mmichaelis** [comment]
@FilipTokarski, thanks for your response. I can confirm: For the docs page it works as well for me (within Chrome).

I tried to strip down the plugins and ended up with:

* Essentials
* Link
* Paragraph

And the effect is the same, i.e. cursor is at the end of the placeholder text.

Thus, either there is some other side-effect or it is the version (we are at 24.0.0, cannot easily check for up-to-date version).

Another observed difference: If we focus our editor, the placeholder disappears. If I focus the editor in CKEditor documentation, the placeholder text remains until you start typing.

**Comment by mmichaelis** [comment]
I assume the new behavior got introduced with #8689. This may already fix the issue, as I think it is ok, that the empty text remains until the link-plugin has added some text, actually. Fix #8689 made it onto master in February 2021, while 24.0.0 is from December 2020.

**Comment by FilipTokarski** [comment]
Thanks for the details. It's exactly as you described, this new behaviour [was introduced in version `26.0.0`](https://github.com/ckeditor/ckeditor5/releases/tag/v26.0.0) and the issue no longer exists. I guess we can close this ticket.

P.S.
There's a useful tip if you want to quickly check something on different versions of the editor - you can switch versions on our docs page. Fox example instead of url:
```
https://ckeditor.com/docs/ckeditor5/latest/features/editor-placeholder.html
```
You can use:
```
https://ckeditor.com/docs/ckeditor5/24.0.0/features/editor-placeholder.html
```
